File: sdk/client/java/openapi-test/src/test/java/opensdk/sdk/apis/klaytnDebug/profiling/DebugSetBlockProfileRateTest.java
package opensdk.sdk.apis.klaytnDebug.profiling;

import opensdk.sdk.apis.constant.UrlConstants;
import opensdk.sdk.models.DebugSetBlockProfileRateResponse;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;
import org.klaytn.OpenSDK;

import java.io.IOException;

@DisplayName("Debug RPC Test")
public class DebugSetBlockProfileRateTest {
    private final OpenSDK sdk = new OpenSDK(UrlConstants.SERVER_URL);

    @Test
    @DisplayName("RPC debug_setBlockProfileRate")
    void whenRequestValid_ThenCall200ResponseReturns() throws IOException {
        int rate = 3;

        DebugSetBlockProfileRateResponse response = sdk.debug.setBlockProfileRate(rate).send();
        response.getResult();
    }
}

File: sdk/client/javascript/openapi-test/test/klaytnDebug/profiling/DebugSetBlockProfileRate.test.js
const OpenSdk = require("opensdk-javascript");
const { expect } = require("@jest/globals");
const { RPC } = require("../../constant");

const sdk = new OpenSdk(new OpenSdk.ApiClient(RPC));

describe('debug_setBlockProfileRate API', () => {
    test('should return debug_setBlockProfileRate', (done) => {

        let callbackOne = function (error, data, response) {
            expect(error).toBeNull();
            expect(data).toBeDefined()
            done();
        };

        const rate = 3;

        sdk.debug.setBlockProfileRate(rate, {}, callbackOne);
    });
});

